internal/versions/versions.go

```diff
@@ -175,42 +171,88 @@ func (h *handler) GetDefaultReleaseImage(cpuArchitecture string) (*models.Releas
 
 // Returns the OsImage entity
 func (h *handler) GetOsImage(openshiftVersion, cpuArchitecture string) (*models.OsImage, error) {
-	versionKey, err := h.getKey(openshiftVersion)
-	if err != nil {
-		return nil, err
-	}
-
-	for _, osImage := range h.osImages {
+	// Find OS images of relevant CPU architecture
+	osImages := funk.Filter(h.osImages, func(osImage *models.OsImage) bool {
+		// Empty implies default CPU architecture
+		if swag.StringValue(osImage.CPUArchitecture) == "" {
+			return cpuArchitecture == "" || cpuArchitecture == common.DefaultCPUArchitecture
+		}
 		if cpuArchitecture == "" {
-			// Empty implies default CPU architecture
 			cpuArchitecture = common.DefaultCPUArchitecture
 		}
-		if *osImage.OpenshiftVersion == versionKey && *osImage.CPUArchitecture == cpuArchitecture {
-			return osImage, nil
-		}
+		return swag.StringValue(osImage.CPUArchitecture) == cpuArchitecture
+	})
+	if funk.IsEmpty(osImages) {
+		return nil, errors.Errorf("The requested CPU architecture (%s) isn't specified in OS images list", cpuArchitecture)
 	}
 
-	return nil, errors.Errorf("The requested OS image for version (%s) isn't specified in OS images list", versionKey)
-}
+	// Search for specified x.y.z openshift version
+	osImage := funk.Find(osImages, func(osImage *models.OsImage) bool {
+		return *osImage.OpenshiftVersion == openshiftVersion
+	})
 
-// Returns the ReleaseImage entity
-func (h *handler) GetReleaseImage(openshiftVersion, cpuArchitecture string) (*models.ReleaseImage, error) {
 	versionKey, err := h.getKey(openshiftVersion)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, releaseImage := range h.releaseImages {
-		if cpuArchitecture == "" {
+	if osImage == nil {
+		// Fallback to x.y version
```

Comment:
This isn't necessarily x.y, right? The first pass tries to match whatever is passed in `openshiftVersion`. This tries to match the key for the release image for ocp version `openshiftVersion` right?

Reply:
So it's actually just matching against x.y version of OS image. I.e. we cover three options here:
1. x.y.z is available
2. only x.y is available
3. if both 1 and 2 are not available - find latest x.y.z

Reply:
Ah because `getKey` is really `inMajorMinorFormat`. Got it.